intersection_transit_maneuvering/include/intersection_transit_maneuvering.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <vector>
+#include <cav_msgs/TrajectoryPlan.h>
+#include <cav_msgs/TrajectoryPlanPoint.h>
+#include <cav_msgs/Plugin.h>
+#include <cav_msgs/Maneuver.h>
+#include <boost/shared_ptr.hpp>
+#include <carma_utils/CARMAUtils.h>
+#include <boost/geometry.hpp>
+#include <carma_wm/Geometry.h>
+#include <cav_msgs/IntersectionTransitLeftTurnManeuver.h>
+#include <cav_msgs/IntersectionTransitRightTurnManeuver.h>
+#include <cav_msgs/IntersectionTransitStraightManeuver.h>
+#include <cav_srvs/PlanTrajectory.h>
+#include <carma_wm/WMListener.h>
+#include <functional>
+#include <unordered_set>
+#include <autoware_msgs/Lane.h>
+#include <ros/ros.h>
+#include <itm_helper.h>
+#include <basic_autonomy/helper_functions.h>
+
+using PublishPluginDiscoveryCB = std::function<void(const cav_msgs::Plugin&)>;
+
+namespace intersection_transit_maneuvering
+{
+
+
+    class IntersectionTransitManeuvering
+    {
+        public:
+        /**
+        * \brief Constructor
+        * 
+        * \param wm Pointer to intialized instance of the carma world model for accessing semantic map data
+        * \param plugin_discovery_publisher Callback which will publish the current plugin discovery state
+        * \param obj Interface object to initialize ros::Service::call
+        */ 
+        IntersectionTransitManeuvering(carma_wm::WorldModelConstPtr& wm, PublishPluginDiscoveryCB plugin_discovery_publisher, std::shared_ptr<Interface> obj);
```

Comment:
The carma_wm::WorldModelConstPtr& should not be passed as reference (drop the &) because this is a smart pointer (shared_ptr) and they should be passed directly to ensure proper reference counting occurs. 
Additionally, passing a pointer by reference has basically no efficiency benefit as a pointer is already a very small element to pass into a method.